### DISCUSSION TITLE
Can NotBeNull() keep the property type?

### DISCUSSION THREAD

**Discussion by mu88** [fluentassertions/fluentassertions#1748] [Q&A] [answered]
Hey guys 👋🏻

The following code does not compile at the moment:
```cs
var testee = new HttpResponseMessage(HttpStatusCode.Accepted);
testee.Should().NotBeNull().And.Subject.StatusCode.Should().Be(HttpStatusCode.Accepted);
```

Is it somehow possible that `NotBeNull()` keeps the property type (here `HttpResponseMessage`) instead of falling back to `object`?

**Comment by dennisdoomen** [comment, marked as answer]
Because of limitations of the compiler, we cannot define a generic `Should` extension method on `T` and still have more specific extension method for types like `IEnumerable<T>`, etc. That's why your example is using the non-generic `ObjectAssertions`. 

There's a rather ugly workaround:

`testee.Should().NotBeNull().And.Subject.As<HttpResponseMessage>().StatusCode.Should().Be(HttpStatusCode.Accepted);`

But my recommendation is to replace this call with:

```
testee.Should().BeEquivalentTo(new
{
    StatusCode = HttpStatusCode.Accepted
});
```

**Reply by mu88** [reply]
Thanks Dennis, got it!
